cmake/Options.cmake

```diff
@@ -15,6 +15,7 @@ option(TP_ENABLE_SHM "Enable shm transport" ${LINUX})
 
 # Channels
 option(TP_ENABLE_CMA "Enable cma channel" ${LINUX})
+option(TP_ENABLE_CUDA_IPC "Enable CUDA IPC channel" ${LINUX})
```

Comment:
Fine for now, but maybe eventually we'll want the default to also take into account whether CUDA can be found or not in the system. Here's what I imagine:
- TP_ENABLE_CUDA_IPC left unspecified by the user: build it if on Linux and CUDA available, don't build it if not, never raise an error
- TP_ENABLE_CUDA_IPC enabled by the user: raise an error if not on Linux or CUDA not found
- TP_ENABLE_CUDA_IPC disabled by the user: don't build it, even if on Linux, even if CUDA found